pallets/pallet-bonded-coins/src/try_state.rs

```diff
@@ -0,0 +1,83 @@
+use frame_support::traits::{
+	fungible::InspectHold,
+	fungibles::{metadata::Inspect as InspectMetadata, roles::Inspect as InspectRoles, Inspect},
+};
+use sp_runtime::{traits::Zero, TryRuntimeError};
+
+use crate::{types::PoolDetails, Config, FungiblesAssetIdOf, HoldReason, Pools};
+
+pub(crate) fn do_try_state<T: Config>() -> Result<(), TryRuntimeError> {
```

Comment:
Should you also add checks for frozen balances?

Reply:
Yeah, I wanted to do that, but this is not easy to achieve. The asset itself is not frozen, only the accounts. I have no way to fetch those accounts and check their state